java/src/main/java/com/logicalclocks/hsfs/TrainingDataset.java

```diff
@@ -423,22 +423,28 @@ public void deleteTag(String name) throws FeatureStoreException, IOException {
 
   @JsonIgnore
   public String getQuery() throws FeatureStoreException, IOException {
-    return getQuery(Storage.ONLINE, false);
+    return getQuery(Storage.ONLINE, false, false);
   }
 
   @JsonIgnore
   public String getQuery(boolean withLabel) throws FeatureStoreException, IOException {
-    return getQuery(Storage.ONLINE, withLabel);
+    return getQuery(Storage.ONLINE, withLabel, false);
   }
 
   @JsonIgnore
   public String getQuery(Storage storage) throws FeatureStoreException, IOException {
-    return getQuery(storage, false);
+    return getQuery(storage, false, false);
   }
-
+  
   @JsonIgnore
   public String getQuery(Storage storage, boolean withLabel) throws FeatureStoreException, IOException {
-    return trainingDatasetEngine.getQuery(this, storage, withLabel);
+    return trainingDatasetEngine.getQuery(this, storage, withLabel, false);
+  }
+
+  @JsonIgnore
+  public String getQuery(Storage storage, boolean withLabel, boolean isHiveQuery)
```

Comment:
Same as for python part, we don't want to surface this to the user and if so he sets it once when creating the connection object already.

Reply:
@moritzmeister In Java client, engine is not specified when creating the connection. Also, hive engine is not available. I think the user need to specify it in this case.

Reply:
ah yes I am stupid, there is not Hive engine in scala. so this is actually irrelevant. no need for changes in the Java part of the client. Sorry about that. you can keep the setting in the lower level classes for the future, but simply always pass False, since it's always Spark.